profile/filter.go

```diff
@@ -41,8 +41,8 @@ func (p *Profile) FilterSamplesByName(focus, ignore, hide, show *regexp.Regexp)
 			}
 		}
 		if show != nil {
-			l.Line = l.matchedLines(show)
-			if len(l.Line) == 0 {
+			if !l.matchesName(show) {
```

Comment:
It feels the new code won't handle correctly show filter when it matches some lines within the location. l.matchesName() will return true in that case which means line filtering won't be done.

Reply:
good catch. i fixed the issue and added test cases to cover it.